stubs/pyinstaller/PyInstaller/building/build_main.pyi

```diff
@@ -0,0 +1,28 @@
+# Referenced in: https://pyinstaller.org/en/stable/hooks.html?highlight=get_hook_config#PyInstaller.utils.hooks.get_hook_config
+# Not to be imported during runtime, but is the type reference for hooks and analysis configuration
+
+from _typeshed import StrOrBytesPath
+from collections.abc import Iterable
+
+from PyInstaller.building.datastruct import Target
+
+class Analysis(Target):
+    # https://pyinstaller.org/en/stable/hooks-config.html#hook-configuration-options
+    hooksconfig: dict[str, dict[str, object]]
+    def __init__(
+        self,
+        scripts: Iterable[StrOrBytesPath],
+        pathex=...,
+        binaries=...,
+        datas=...,
+        hiddenimports=...,
+        hookspath=...,
+        hooksconfig: dict[str, dict[str, object]] | None = ...,
+        excludes=...,
+        runtime_hooks=...,
+        cipher=...,
+        win_no_prefer_redirects: bool = ...,
+        win_private_assemblies: bool = ...,
+        noarchive: bool = ...,
+        module_collection_mode=...,
+    ) -> None: ...
```

Comment:
I'm confused about why pyright isn't emitting an error here. There's no entry in the `pyrightconfig.stricter.json` exclude-list, so it should be erroring for any parameters that don't have a type annotation.

Reply:
Just a guess:
`pyrightconfig.stricter.json` has `"typeCheckingMode": "basic"`
and doesn't configure `reportMissingParameterType`. The default value for that setting being `None`

Reply:
No idea -- anyway, it's not your fault, and we certainly don't have a requirement that all parameters need to be annotated when first adding stubs :) I'll look into it and try to work it out...